source/catalog/controller/CatalogServiceController.ts

```diff
@@ -144,7 +144,9 @@ export default class CatalogServiceController {
         // At this point, all relevant items should filtered.  If picking by version, choose the latest that matches
         if (Version) {
             // Keep versions that satisfy the passed in Version.
-            filteredCandidates = filteredCandidates.filter((item) => semver.satisfies(item.Version, Version));
+            filteredCandidates = filteredCandidates.filter((item) =>
+                (semver.valid(Version) && semver.valid(item.Version) && semver.eq(item.Version, Version))
```

Comment:
If I am reading this right, this filters down to either the specified version exactly, the semverange, or both - right?

Reply:
and just to add on to that, can we do this filtering and sorting when we are doing the query instead of post processing?

Reply:
Yep!

Reply:
Technically it is both, but in the end, if there is an exact match, it will be the one picked.

Reply:
@sairathb I don't think it's possible while retaining the semver package and all this custom logic. It is happening immediately after the query though:

![image](https://user-images.githubusercontent.com/15173843/57717710-7175ad00-7641-11e9-8067-c5708c8408b5.png)

Reply:
@sairathb The version checking is done by the semver lib, and trying to move that into a query would get super complicated.  We're also doing it in a very similar way to the Cloud Map service.

Reply:
The way I was imagining it was to do a basic filtering on the given version number in the query and then applied semver on it. For example, if the user provides a version 1, then I add that as a beginsWith condition on version in the query. Once I get back the results, if it is more than one, I can do a semver, otherwise return the record. If 0 records are returned, then throw an exception. This takes care of full version Vs partial versions. @TwoSlick @mattcookxyz